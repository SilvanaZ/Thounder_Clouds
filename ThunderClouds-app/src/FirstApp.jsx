import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle, name }) => {
  //console.log(props);

  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      {/*<code>{JSON.stringify(newMessage)}</code>*/}
      <p>{name}</p>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
    </>
  );
};

/* Una validación de los accesorios que se pasan al componente. */
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

/* Un valor predeterminado para los accesorios. */
FirstApp.defaultProps = {
  name: "Rocío Fernandez",
  subTitle: "No hay subtitulos",
  title: "No hay titulo",
};
