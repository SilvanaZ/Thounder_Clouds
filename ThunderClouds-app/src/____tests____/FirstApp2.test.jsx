import { getAllByText, render, screen } from '@testing-library/react'
import { FirstApp } from "../FirstApp";

describe('Pruebas en <FirstApp2/>', () => {

    const title = 'Hola, soy Pelufo';
    const subTitle = 'Ac la manosera, escribiendo el subtitulo';

    test('it must make mach with the snapshot', () => {

        const { container } = render(<FirstApp title={title} />)
        expect(container).toMatchSnapshot();
    });
    test('debe mostrar un mensaje "Hola, soy Pelufo"', () => {

        render(<FirstApp title={title} />)
        //screen.debug();
        expect(screen.getByText(title)).toBeTruthy();
    });
    test('debe mostrar el titulo en h1', () => {

        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title)

    });
    test('debe mostrar subtitulo enviado por props', () => {

        render(<FirstApp
            title={title}
            subTitle={subTitle}

        />
        );
        expect(screen.getAllByText(subTitle).length).toBe(2);

    });

});
