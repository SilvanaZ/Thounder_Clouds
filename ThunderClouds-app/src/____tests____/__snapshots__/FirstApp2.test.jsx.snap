// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp2/> it must make mach with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Pelufo
  </h1>
  <p>
    Roc√≠o Fernandez
  </p>
  <p>
    No hay subtitulos
  </p>
  <p>
    No hay subtitulos
  </p>
</div>
`;
